quiche/src/lib.rs

```diff
@@ -8196,8 +8323,18 @@ impl TransportParams {
                     tp.max_datagram_frame_size = Some(val.get_varint()?);
                 },
 
-                // Ignore unknown parameters.
-                _ => (),
+                // Track unknown transport parameters specially.
+                unknown_tp_id =>
+                    if track_unknown_parameters {
+                        // It is _not_ an error not to have space enough to track
+                        // an unknown parameter.
+                        let _track_unknown_parameter_result = tp
+                            .unknown_params
+                            .safe_push(UnknownTransportParameter {
+                                id: unknown_tp_id,
+                                value: val.to_vec(),
```

Comment:
IIUC this will still require an allocation even if there is no capacity. Yes, there wouldn't be unbound memory growth, but it could still consume needless cycles. 

An alternative would be to change the signature of safe_push to be `id: u64, value: &[u8]`, then inside it you can create UnknownTransportParameter with a vec copy, only if you know there's capacity. You can use `val.as_ref()` to get the ref to pass

Reply:
I just want to confirm what I think you are saying: The `to_vec()` is going to consume cycles and (potentially) cause an allocation just to pass it to the `safe_push` member function and, even after all that work, it may not even use it!

If I am hearing what you are saying, then I absolutely agree and will make the change!

Reply:
Yep!

Reply:
10-4 -- I am on it!

Reply:
I have a pretty "cool" implementation of this. I want to look over it a little more before I send you another revision of this PR. Sorry for the delay and thank you for the feedback!